### PR TITLE
gconf: add dependencies

### PR DESCRIPTION
**Pull request by ketsubouchi**
Add dependencies (See #18400 and #18405 ) and delete TODO.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gconf/package.py [hunk 1]
```diff
@@ -14,13 +14,10 @@ class Gconf(AutotoolsPackage):
...
-    # gio-2.0 >= 2.31.0
-    # gthread-2.0
-    # gmodule-2.0 >= 2.7.0
-    # gobject-2.0 >= 2.7.0
-    # dbus-1 >= 1.0.0
-    # dbus-glib-1 >= 0.74
+    depends_on('dbus')
+    depends_on('dbus-glib')
+    depends_on('orbit2')
+    depends_on('perl-xml-parser')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gconf/package.py]: Is this type=build or type=build/run?

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/gconf/package.py]: Considering functions, all new packages seems to be "type=build/run".

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gconf/package.py]: The default is build/link. I don't think this Perl package even installs a library, right?

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/gconf/package.py]: I was confused. ”build/link" seems enough in this case.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gconf/package.py]: But we don't do any linking, right? In that case, it should be `type='build'`

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/gconf/package.py]: I checked installed files again.
Path to `perl` package is included in `gconf` installed files such as `bin/gconftool-2`.
So when gconf is run, perl and xml-parser will be called.
I think type = 'build/run' is needed for perl-xml-parser, others are default.

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/gconf/package.py]: Updated patch

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**